src/engine/InputManager.java

```diff
@@ -86,16 +101,21 @@ public void keyReleased(final KeyEvent key) {
 	 * @return True if the key was double-tapped, false otherwise.
 	 */
 	public boolean isDoubleTap(int keyCode) {
+		if (keyCode != KeyEvent.VK_LEFT && keyCode != KeyEvent.VK_RIGHT) { return false; }
 		long currentTime = System.currentTimeMillis();
 		long lastPressTime = lastKeyPress.getOrDefault(keyCode, 0L);
-		boolean isDoubleTap = (currentTime - lastPressTime) <= DOUBLE_TAP_THRESHOLD;
-		lastKeyPress.put(keyCode, currentTime);
+		boolean isDoubleTap = (currentTime - lastDoubleTapTime > DOUBLE_TAP_COOLDOWN) &&
+				(currentTime - lastPressTime > 0) &&
+				(currentTime - lastPressTime <= DOUBLE_TAP_THRESHOLD);
```

Comment:
This section detect key press in 300 milliseconds instead of double tap.
I think this line will be useful in KeyPressed function.